packages/device-client/src/lib/services/index.ts

```diff
@@ -0,0 +1,488 @@
+import { Config } from '@forgerock/javascript-sdk';
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import { DeleteOathQuery, OathDevice, OAthResponse, RetrieveOathQuery } from '../types/oath';
+import { GetSessionInfoQuery, GetSessionInfoResponse } from '../types/getSessionInfo';
+import { DeleteDeviceQuery, PushDeviceQuery, PushDevicesResponse } from '../types/push-device';
+import { BindingDeviceQuery, DeviceResponse } from '../types/binding-device';
+
+import {
+  WebAuthnBody,
+  WebAuthnCredential,
+  WebAuthnDevicesResponse,
+  WebAuthnQuery,
+  WebAuthnQueryWithUUID,
+} from '../types/webauthn';
+
+export const deviceService = () =>
+  createApi({
+    reducerPath: 'deviceClient',
+    baseQuery: fetchBaseQuery({
+      credentials: 'include',
+      prepareHeaders: (headers) => {
+        headers.set('Content-Type', 'application/json');
+        headers.set('Accept', 'application/json');
+        headers.set('x-requested-with', 'forgerock-sdk');
+        headers.set('x-requested-platform', 'javascript');
+        return headers;
+      },
+      baseUrl: Config.get().serverConfig.baseUrl,
+    }),
+    endpoints: (builder) => ({
+      retrieveSessionInfo: builder.mutation<
+        GetSessionInfoResponse,
+        GetSessionInfoQuery | undefined
+      >({
+        async queryFn(query = { realm: '' }, api, _options, baseQuery) {
```

Comment:
If you didn't want to write up the logic for all of these instances of `queryFn`, you could just pass in the realm as a argument, and then just use `query`. That would potentially remove 90% of this code.

Reply:
Gonna send msg on slack